src/back/wgsl/writer.rs

```diff
@@ -632,8 +632,15 @@ impl<W: Write> Writer<W> {
                 write!(self.out, "{}", INDENT.repeat(indent))?;
                 writeln!(self.out, "discard;")?
             }
-            // TODO: copy-paste from glsl-out
             Statement::Store { pointer, value } => {
+                // WGSL does not support all SPIR-V builtins and we should skip it in generated shaders.
```

Comment:
this is unfortunate... We shouldn't be in this situation, ideally.
If backends aren't expected to handle a particular builtin, the validator should catch this. And it would be responsibilty of spv-in to filter those out,

Reply:
So, it means that `spv-in` should know about backend restrictions (capabilities?) and parse all that backend can handle and skip all other stuff?

What if the backend can emulate unsupported things somehow? Should I open a new issue for this?

Reply:
No, spv-in wouldn't need to know about backends. It would need to know about what IR expects. IR shouldn't blindly pass everything under the sun, it recently got a bitset of capabilities itself, to validate. One of them is push constants, for example. So I imagine `spv-in` could have some options to omit particular things from the IR, and the user would provide them in tandem with specifying the validation capabilities for the IR.